### DISCUSSION TITLE
502 gateway when sending a request to live site

### DISCUSSION THREAD

**Discussion by thomas159** [vercel/next.js#19446] [Help]
I have just deployed my site to netlify and I am getting a 502 error when a request hits my `pages/api/hook.js` api , it works fine locally and returns a 200 status. 

I have pasted the headers incase there is something I need to set on the api with headers?

**hook..js**
```
const db = require('quick.db')

export default async (req, res) => {
  const { method, body } = req
  if (method === 'POST') {
   
    console.log('userInfo', db.get('userInfo'))
    res.statusCode = 200
    res.end()
  } else if (method === 'GET') {
    db.set('userInfo', req.cookies.tagrid)
    console.log('fish', db.get('userInfo'))
    res.statusCode = 200
    res.end()
  }
}

```
**Headers.js**
`Request URL: https://relaxen-test2.netlify.app/api/hook
Request Method: GET
Status Code: 502 
Remote Address: [2a03:b0c0:3:e0::1b:1]:443
Referrer Policy: strict-origin-when-cross-origin
age: 1
cache-control: no-cache
content-length: 586
content-type: text/plain; charset=utf-8
date: Mon, 23 Nov 2020 15:46:24 GMT
server: Netlify
status: 502
x-nf-request-id: c9cd5fb7-dfb0-4f8c-8114-3db0a12e5799-8915950
:authority: relaxen-test2.netlify.app
:method: GET
:path: /api/hook
:scheme: https
accept: */*
accept-encoding: gzip, deflate, br
accept-language: en,de-DE;q=0.9,de;q=0.8,en-US;q=0.7
cache: no-cache
content-type: application/text
cookie: _ga=GA1.3.720796213.1604759790; _fbp=fb.2.1604759789876.532304367; Relaxen=true; _gid=GA1.3.403617601.1606130086; tagrid=33843
credentials: include
mode: no-cors
referer: https://relaxen-test2.netlify.app/
sec-fetch-dest: empty
sec-fetch-mode: cors
sec-fetch-site: same-origin
user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36`

**response error:**

`{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"Error: Could not find module root given file: \"/var/task/src/nextPage.js\". Do you have a `package.json` file? ","trace":["Runtime.UnhandledPromiseRejection: Error: Could not find module root given file: \"/var/task/src/nextPage.js\". Do you have a `package.json` file? ","    at process.<anonymous> (/var/runtime/index.js:35:15)","    at process.emit (events.js:314:20)","    at processPromiseRejections (internal/process/promises.js:209:33)","    at processTicksAndRejections (internal/process/task_queues.js:98:32)"]}`

**Comment by Tomburgs** [comment]
The problem you're facing is that Netlify is not a web server, it merely generates and distributes your files through CDN.

Since you're using some kind of database it will not work as your code is executed client-side.
